internal/publish/database/exposure.go

```diff
@@ -261,40 +264,106 @@ func (db *PublishDB) ReadExposures(ctx context.Context, b64keys []string) (map[s
 	return result, nil
 }
 
+func prepareInsertExposure(ctx context.Context, tx pgx.Tx) (string, error) {
```

Comment:
Is there a reason we don't use

```
INSERT INTO Exposure ...
ON CONFLICT (exposure_key)
DO
  UPDATE EXPOSURE SET ...
```

? Seems like it would simplify this and ensure atomicity? Right now there's a race where the exposure could be created between the time you look it up and the time you try to create it.

Reply:
Stackover for posterity: https://stackoverflow.com/questions/1109061/insert-on-duplicate-update-in-postgresql

Reply:
moved revision entirely in a DB transaction.